library/cipher_wrap.c

```diff
@@ -583,22 +585,22 @@ static int gcm_aes_setkey_wrap(void *ctx, const unsigned char *key,
 static const mbedtls_cipher_base_t gcm_aes_info = {
     MBEDTLS_CIPHER_ID_AES,
     NULL,
-#if defined(MBEDTLS_CIPHER_MODE_CBC)
+#if defined(MBEDTLS_CIPHER_MODE_CBC) && defined(MBEDTLS_CIPHER_C)
     NULL,
 #endif
-#if defined(MBEDTLS_CIPHER_MODE_CFB)
+#if defined(MBEDTLS_CIPHER_MODE_CFB) && defined(MBEDTLS_CIPHER_C)
     NULL,
 #endif
-#if defined(MBEDTLS_CIPHER_MODE_OFB)
+#if defined(MBEDTLS_CIPHER_MODE_OFB) && defined(MBEDTLS_CIPHER_C)
     NULL,
 #endif
-#if defined(MBEDTLS_CIPHER_MODE_CTR)
+#if defined(MBEDTLS_CIPHER_MODE_CTR) && defined(MBEDTLS_CIPHER_C)
     NULL,
 #endif
-#if defined(MBEDTLS_CIPHER_MODE_XTS)
+#if defined(MBEDTLS_CIPHER_MODE_XTS) && defined(MBEDTLS_CIPHER_C)
     NULL,
 #endif
-#if defined(MBEDTLS_CIPHER_MODE_STREAM)
+#if defined(MBEDTLS_CIPHER_MODE_STREAM) && defined(MBEDTLS_CIPHER_C)
     NULL,
 #endif
 #if defined(MBEDTLS_GCM_C)
```

Comment:
I think this should have `&& defined(MBEDTLS_CIPHER_C)` as well.

Reply:
That holds for every instance `GCM_C`, `CCM_C` or `MBEDTLS_CIPHER_HAVE_GCM_AES_VIA_LEGACY_OR_USE_PSA` (whose definition we probably want to change to include the `CIPHER_C` bit, rather than adding it to all occurrences of this macro).

Reply:
(Note: there are instances in `cipher.c` which are concerned as well.)